### ISSUE TITLE
TestRestClientInvocation的单元测试无法有效验证功能

### ISSUE THREAD

**Issue by yhs0092** [apache/servicecomb-java-chassis#3739]
问题出在 `TestRestClientInvocation` 的 `testSetCseContext_enable_unicode` 方法和 `testSetCseContext_disable_unicode` 方法中, 两个方法都是开头对`invocation.getContext()`进行mock, 结尾又调用`invocation.getContext()`获取结果进行校验, 获取的结果就是mock返回, 所以达不到测试的预期效果. 1.3.x 和 2.8.x 分支均有此问题, 建议优化一下UT代码.

https://github.com/apache/servicecomb-java-chassis/blob/21d4b4863b71332f151eb63020f4e9ac51583002/transports/transport-rest/transport-rest-client/src/test/java/org/apache/servicecomb/transport/rest/client/http/TestRestClientInvocation.java#L282-L346

**Comment by lbc97** [comment]
这个测试本身我试过了是可以通过的 我没发现问题在哪儿 你是说这个测试方法本身要验证的点就是不对的吗

**Comment by yhs0092** [comment]
> 你是说这个测试方法本身要验证的点就是不对的吗

是的, `invocation.getContext()` 方法本身被 mock 了, 所以你在单元测试方法最后取的测试值是mock数据`contextMap`里面的值, 而非UT中经过计算转换的值.

你可以试一下把`org.apache.servicecomb.transport.rest.client.http.RestClientInvocation#setCseContext` 和 `org.apache.servicecomb.common.rest.AbstractRestInvocation#setContext`都注释掉, 不会影响你这个单元测试的结果.
